utils/xbox.ixx

```diff
@@ -0,0 +1,125 @@
+module;
+#include <algorithm>
+#include <cstdint>
+#include <vector>
+
+export module utils.xbox;
+
+
+
+namespace gpsxre
+{
+
+export enum class XGD_Type : uint8_t
+{
+    UNKNOWN,
+    XGD1,
+    XGD2,
+    XGD3
+};
+
+export XGD_Type get_xgd_type(const std::vector<uint8_t> &ss)
+{
+    if(ss.size() != 2048)
+        return XGD_Type::UNKNOWN;
+
+    // Concatenate the last three values
+    const uint32_t xgd_type = ((uint32_t)ss[13] << 16) | ((uint32_t)ss[14] << 8) | ss[15];
+
+    // Return XGD type based on value
+    switch(xgd_type)
+    {
+    case 0x2033AF:
+        return XGD_Type::XGD1;
+    case 0x20339F:
+        return XGD_Type::XGD2;
+    case 0x238E0F:
+        return XGD_Type::XGD3;
+    default:
+        return XGD_Type::UNKNOWN;
+    }
+}
+
+export void clean_xbox_security_sector(std::vector<uint8_t> &ss)
```

Comment:
I have found this to be a helpful resource for XGD1 security sectors (I believe it is mostly the same for XGD2 and 3 but there are some differences) but I don't think it really covers the memory regions being checked here
https://xboxdevwiki.net/Xbox_Game_Disc#Security_Sectors_.28SS.bin.29

This specific block of code is coming from the MPF processing code, which I believe was based off of the redump community sector_ranges exe file that was shared.

@Deterous do you have any other resources regarding this?

Reply:
I've been working on my own description of the structure as info on the internet is sorely lacking (above link seems to be mostly it, old sites that may have had info have been taken down).
We only use one  component of the media_specific portion, so for now we can replace all instances of `std::vector` with `READ_DVD_STRUCTURE_LayerDescriptor`

Reply:
I think it can be accepted as is for the time being. At some point I'd be interested to try kreon myself and understand more the structures, but that will happen later.